Maple2.Server.Game/Manager/GroupChatManager.cs

```diff
@@ -0,0 +1,181 @@
+
+        var groupChat = new GroupChat(info.Id);
+        foreach (GroupChatMember member in members) {
+            if (groupChat.Members.TryAdd(member.CharacterId, member)) {
+                BeginListen(member);
+            }
+        }
+
+        this.groupChat = groupChat;
+
+        session.Send(GroupChatPacket.Load(this.groupChat));
+        return true;
+    }
+
+    public bool AddMember(GroupChatMember sender, GroupChatMember receiver) {
+        if (groupChat == null) {
+            return false;
+        }
+        if (!groupChat.Members.TryAdd(receiver.CharacterId, receiver)) {
+            return false;
+        }
+
+        BeginListen(receiver);
+        session.Send(GroupChatPacket.AddMember(receiver.Info, sender.Name, groupChat.Id));
+        return true;
+    }
+
+    public bool RemoveMember(long characterId) {
+        if (groupChat == null) {
+            return false;
+        }
+        if (!groupChat.Members.TryRemove(characterId, out GroupChatMember? member)) {
+            return false;
+        }
+        EndListen(member);
+
+        session.Send(characterId == session.CharacterId ? GroupChatPacket.Leave(groupChat.Id) :
+            GroupChatPacket.RemoveMember(member.Name, groupChat.Id));
+        return true;
+    }
+
+
+    public bool Disband() {
+        if (groupChat == null) {
+            return false;
+        }
+
+        foreach (GroupChatMember member in groupChat.Members.Values) {
+            EndListen(member);
+        }
+
+        groupChat = null;
+        return true;
+    }
+
+    public void CheckDisband() {
+        if (groupChat == null) {
+            return;
+        }
+
+        // Checking less or equal to 1 because member logging off is still online at this point
+        if (groupChat.Members.Values.Count(member => member.Info.Online) <= 1) {
+            session.World.GroupChat(new GroupChatRequest {
+                Disband = new GroupChatRequest.Types.Disband {
+                    GroupChatId = groupChat.Id,
+                },
+                RequestorId = session.CharacterId,
+            });
+        }
+    }
+
+    public GroupChatMember? GetMember(long characterId) {
+        if (groupChat?.Members.TryGetValue(characterId, out GroupChatMember? member) == true) {
+            return member;
+        }
+
+        return null;
+    }
+
+    #region PlayerInfo Events
+    private void BeginListen(GroupChatMember member) {
+        // Clean up previous token if necessary
+        if (member.TokenSource != null) {
+            logger.Warning("BeginListen called on Member {Id} that was already listening", member.CharacterId);
+            EndListen(member);
+        }
+
+        member.TokenSource = CancellationTokenSource.CreateLinkedTokenSource(tokenSource.Token);
+        CancellationToken token = member.TokenSource.Token;
+        var listener = new PlayerInfoListener(UpdateField.Party, (type, info) => SyncUpdate(token, member.CharacterId, type, info));
```

Comment:
`UpdateField.Party` is probably wrong here, you only want to subscribe to the update events that matter to group chat.